libblkid: (gpt) use real sector size to set PTMAGIC_OFFSET

the patch comes from util-linux
(git://git.kernel.org/pub/scm/utils/util-linux/util-linux.git):
commit 8d8a02398a01d8d2930676c992bc91d7ceb8bc17

So stupid and 5 years old bug...

Signed-off-by: Karel Zak <kzak@redhat.com>
Signed-off-by: Li Wang <li.wang@windriver.com>
---
 libblkid/src/partitions/gpt.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libblkid/src/partitions/gpt.c b/libblkid/src/partitions/gpt.c
index 7fccd51..728b6ef 100644
--- a/libblkid/src/partitions/gpt.c
+++ b/libblkid/src/partitions/gpt.c
@@ -329,7 +329,7 @@ static int probe_gpt_pt(blkid_probe pr,
 
 	blkid_probe_use_wiper(pr, lba * blkid_probe_get_size(pr), 8);
 
-	if (blkid_probe_set_magic(pr, lba << 9,
+	if (blkid_probe_set_magic(pr, blkid_probe_get_sectorsize(pr) * lba,
 			      sizeof(GPT_HEADER_SIGNATURE_STR) - 1,
 			      (unsigned char *) GPT_HEADER_SIGNATURE_STR))
 		goto err;
-- 
1.7.9.5

